### DISCUSSION TITLE
can not add the azure dalle-3 tool

### DISCUSSION THREAD

**Discussion by waterzhong2023** [langgenius/dify#5661] [General]
I just cant no configure the tool for AZURE AI DALLE-3, but it is working with python sdk. could you help ？

![image](https://github.com/langgenius/dify/assets/129365638/0d85a52c-b966-425c-adb2-122bd82059c8)

![image](https://github.com/langgenius/dify/assets/129365638/46996cfe-a27b-42b9-838f-1cf19e712a3b)

**Comment by keita** [comment]
Have you tried setting the API Version to 2023-12-01-preview? It's working for me in my self host environment.

**Reply by epam-vendor** [reply]
still the same, error : no route, here is my setting
![image](https://github.com/langgenius/dify/assets/167284158/a0ee82b1-2733-4520-a4c1-90930c11a590)

could you show me you setting ? thanks.

**Reply by keita** [reply]
I have this set up in my environment. This was configured using version 0.6.10.

![image](https://github.com/langgenius/dify/assets/5487/d5315940-4407-4226-99a9-d2a6514abeca)

I retried configuring with 0.6.11, but it failed with a different error(name 'random' is not defined). It appears to be the same issue as #5281.

![image](https://github.com/langgenius/dify/assets/5487/6e123386-0a9f-4695-a3d2-b72358eff297)

**Reply by epam-vendor** [reply]
great，a bug； anyway，i will defind a workflow for it，i dont want to rollback my version now。thanks。

**Reply by keita** [reply]
As a heads up, I tried configuring with 0.6.12-fix1 in my environment and was able to successfully authenticate and generate images.

**Reply by epam-vendor** [reply]
sadly , I got a 'No route' error , even with 0.6.12-fix1
![Screenshot_1](https://github.com/langgenius/dify/assets/167284158/71d0f4f6-fcfa-4b0f-a932-d4f1bd179b85)
![Screenshot_2](https://github.com/langgenius/dify/assets/167284158/e289a135-0950-4908-a887-38597d351ce0)
